### PR TITLE
add parseDate helper to fix edge case

### PR DESCRIPTION
**Pull request by yusefnapora**
@Gozala I'm writing tests for my js pinning client and hit this, since my test pins are all created within a few ms of each other.

Apparently calling `Date.parse(anExistingDateInstance)` will throw away the milliseconds part of the input `Date`. It also seems like by the time the `match` function is called, something has already converted the `query.before` and `query.after` params to `Date` objects.

This adds a `parseDate` helper to check if we've already got a `Date` and cast it to an ISO string before calling `Date.parse()`.

Thanks javascript...

### CODE REVIEW

File: service/pins.js [hunk 1]
```diff
@@ -229,6 +229,22 @@ const deriveStatus = ({ name = "" }) => {
...
+ * 
+ * @param {Date|string} d - a Date object or ISO-formatted date string.
+ * @returns {number} - the number of milliseconds elapsed since January 1, 1970 00:00:00 UTC
+ */
+const parseDate = (d) => {
+  if (d instanceof Date) {
+    // Calling Date.parse on a Date instance causes the milliseconds to be thrown away,
+    // so e.g. a date representing 1616522896307 becomes 1616522896000.
+    // Calling toISOString and parsing _that_ seems to work. 
+    return Date.parse(d.toISOString())
```
Comment (Gozala) [inline service/pins.js:243]: Could not this just do ?

```suggestion
    return d
```

**Review by Gozala — APPROVED** [review]

File: service/pins.js (hunk 1 above)
Comment (yusefnapora) [reply service/pins.js:243]: Huh, I guess so. I did the awkward string cast because `Date.parse` returns a number instead of a `Date` object & it seemed more consistent... but apparently you can compare `Date` and numbers using `<`, etc and it does the right thing.
